Packaging

kot-begemot-uk opened the pull request.

kot-begemot-uk committed:
Fix regression in relay code

A regression to the relay code was introduced as a result
of adding some of the agent/mitm code. This commit fixes it.

Additionally, this commit adds debug and tracing to the cps code
paths.

Signed-off-by: Anton Ivanov <anton.ivanov@cambridgegreys.com>

kot-begemot-uk committed:
Add checks to prevent possible agent exits on no ip address

If an interface is unconfigured, the agent will exit when
trying to use it.

This commit adds some checks to prevent the most common cases.

Signed-off-by: Anton Ivanov <anton.ivanov@cambridgegreys.com>

kot-begemot-uk committed:
Address comments on style, naming and fix descriptions

Signed-off-by: Anton Ivanov <anton.ivanov@cambridgegreys.com>

kot-begemot-uk committed:
Merge pull request #1 from open-switch/master

Fix regression in relay code (#2)

kot-begemot-uk committed:
Fixed and amended documentation

Signed-off-by: Anton Ivanov <anton.ivanov@cambridgegreys.com>

kot-begemot-uk committed:
DHCP Agent Packaging

Add debian information and packaging instructions to the dhcp
agent

Signed-off-by: Anton Ivanov <anton.ivanov@cambridgegreys.com>

spenced commented on README.md:
```diff
@@ -9,10 +9,19 @@ The DHCP agent supports two modes of operation.
+ISC DHCP will *NOT* reply to relay requests if it cannot route to the relay agent ip address. This means that a subnet configured for relay mode must be reachable
+from the DHCP server. I would not be surprised if other servers show similar behaviour.
+
+In theory, in order for them to work they will know everything about an incoming
```
Can you de-pronoun this paragraph? It's not exactly clear what you mean.

spenced commented on README.md:
```diff
@@ -36,12 +45,75 @@ Invokes the agent. Unless mock test/mode is specified the agent will register wi
 
-Specifying the optional file argument prevents dhcp from registering with CPS and enables the optional mock mode. In mock
-mode configuration is read in json format from a file and reloaded each time the agent receives SIGUSR1.
+The optional file argument provides a configuration in json format from a file. If the file is specified, it will be reloaded each time the agent receives SIGUSR1.
```
Why not SIGHUP? Does OPX use that for something else?

spenced commented on README.md:
```diff
@@ -36,12 +45,75 @@ Invokes the agent. Unless mock test/mode is specified the agent will register wi
+
+1. Configure a vlan interace using the [VLAN application examples](https://github.com/open-switch/opx-docs/wiki/VLAN-application-examples).
+2. The VLAN must have at least one port for Layer3/Relay mode and at least 2 ports for Layer2/Snooping mode.
+3. For Layer3 mode, the vlan (br100) interface must have an ip address assigned. This can eb done either directly or using [IP address application examples](https://github.com/open-switch/opx-docs/wiki/IP-address-application-examples)
```
Typo: eb -> be

spenced commented on README.md:
```diff
@@ -36,12 +45,75 @@ Invokes the agent. Unless mock test/mode is specified the agent will register wi
+
+##### DHCP Agent application examples
+
+Setup the environment (should not be needed once it is packaged). From the top directory of the source run ``export PYTHONPATH=$PYTHONPATH:`pwd` ``
```
Does this packaging PR remove the need for `export`?

spenced commented on README.md:
```diff
@@ -36,12 +45,75 @@ Invokes the agent. Unless mock test/mode is specified the agent will register wi
+
+Setup the environment (should not be needed once it is packaged). From the top directory of the source run ``export PYTHONPATH=$PYTHONPATH:`pwd` ``
+
+Run the agent `./inocybe_dhcp/opx_dhcp.py --verbose 1`
```
Can the agent be run manually if the service package is installed?

spenced commented on debian/control:
```diff
@@ -0,0 +1,12 @@
+Source: opx-dhcp-agent
```
Does this package ship the DHCP libs? If so, should they be in a separate package?
[debian policy](https://www.debian.org/doc/packaging-manuals/python-policy/module_packages.html)

kot-begemot-uk committed:
Address David Spence comments

1. Rewrite caveats section in the documentation
2. Split the package into a lib and binary

Signed-off-by: Anton Ivanov <anton.ivanov@cambridgegreys.com>

kot-begemot-uk committed:
Rearrange directory structure and naming

Signed-off-by: Anton Ivanov <anton.ivanov@cambridgegreys.com>

kot-begemot-uk committed:
Merge branch 'master' into master
